[Dialogs] Add tests for Dynamic Type.

romoore opened the pull request:
Adds unit and snapshot tests for Dynamic Type support using both MDCTypography
and MDCFontScaler.

Part of #5418

material-automation commented:
bazel detected changes to the following targets:

```
//components/Dialogs:snapshot_tests
//components/Dialogs:snapshot_tests_test_bundle
//components/Dialogs:unit_tests
//components/Dialogs:unit_tests_IPAD_PRO_12_9_IN_9_3
//components/Dialogs:unit_tests_IPAD_PRO_12_9_IN_9_3_test_bundle
//components/Dialogs:unit_tests_IPHONE_7_PLUS_IN_10_3
//components/Dialogs:unit_tests_IPHONE_7_PLUS_IN_10_3_test_bundle
//components/Dialogs:unit_tests_environment
//components/Dialogs:unit_tests_environment_test_bundle
```

<!-- identifier: affected-targets -->

Commit:
[Dialogs] Add tests for Dynamic Type.

Adds unit and snapshot tests for Dynamic Type support using both MDCTypography
and MDCFontScaler.

Part of #5418

Commit:
Trigger view layout to get fonts to resize.

Commit:
Fix snapshots to use default sizing.
